projects/Rockchip/bootloader/firmware

```diff
@@ -7,6 +7,14 @@ else
   export UBOOT_TARGET=""
 fi
 
+if [ -f "$(get_install_dir rkbin)/.noinstall/ddr.bin" ]; then
+  export ROCKCHIP_TPL="$(get_install_dir rkbin)/.noinstall/ddr.bin"
+fi
+
+if [ -f "$(get_install_dir rkbin)/.noinstall/bl31.elf" ]; then
+  export BL31="$(get_install_dir rkbin)/.noinstall/bl31.elf"
+fi
```

Comment:
I find you don't set  `rockchip_legacy_boot = 1` , but set `ROCKCHIP_TPL` and `BL31`.

thus not need:
```
  ${PKG_RKBIN}/tools/mkimage -n "${PKG_SOC}" -T rksd -d "${PKG_DDR_BIN}":"${PKG_MINILOADER}" idbloader.img.rk || exit 1
  ${PKG_RKBIN}/tools/loaderimage --pack --uboot u-boot-dtb.bin uboot.img.rk "${PKG_LOAD_ADDR}" || exit 1

  dd if=idbloader.img.rk of="${UBOOT_FIT_IMAGE}" seek=0 conv=fsync,notrunc > /dev/null 2>&1 || exit 1
  dd if=uboot.img.rk of="${UBOOT_FIT_IMAGE}" seek=16320 conv=fsync,notrunc > /dev/null 2>&1 || exit 1

  if [ "${TARGET_ARCH}" = "aarch64" -o "${TARGET_KERNEL_ARCH}" = "arm64" ]; then
    PKG_ATF_INI="${PKG_RKBIN}"/RKTRUST/"${DEVICE}"TRUST.ini
    if [ ! -f "${PKG_ATF_INI}" ]; then
      echo "install: ${PKG_ATF_INI} does not exist"
      exit 1
    fi

    "${PKG_RKBIN}"/tools/trust_merger --ignore-bl32 --prepath "${PKG_RKBIN}"/ "${PKG_ATF_INI}" || exit 1
    dd if=trust.img of="${UBOOT_FIT_IMAGE}" seek=24512 conv=fsync,notrunc > /dev/null 2>&1 || exit 1
  fi
```
in `install` script.
so `u-boot-rockchip.bin` is directly build from u-boot source, no other step?

Reply:
Yes, the `rockchip_legacy_boot` flag is used for tv-boxes using older generation SoCs.
It means use rk miniloader image instead of plain `u-boot-rockchip.bin`.

For more simple U-Boot build and flashing instructions see U-Boot docs and/or readme and workflows at
https://github.com/Kwiboo/u-boot-build